docs/diagrams/SearchSequenceDiagram.puml

```diff
@@ -0,0 +1,80 @@
+@startuml
+!include style.puml
+
+box Logic LOGIC_COLOR_T1
+participant ":LogicManager" as LogicManager LOGIC_COLOR
+participant ":GradPadParser" as GradPadParser LOGIC_COLOR
+participant ":SearchCommandParser" as SearchCommandParser LOGIC_COLOR
+participant ":ParserUtil" as ParserUtil LOGIC_COLOR
+participant "d:SearchCommand" as SearchCommand LOGIC_COLOR
+participant ":ModuleInfoSearcher" as ModuleInfoSearcher LOGIC_COLOR
+participant ":CommandResult" as CommandResult LOGIC_COLOR
+end box
+
+[-> LogicManager : execute("search CS2103T")
+activate LogicManager
+
+LogicManager -> GradPadParser : parseCommand("search CS2103T")
+activate GradPadParser
+
+create SearchCommandParser
+GradPadParser -> SearchCommandParser
+activate SearchCommandParser
+
+SearchCommandParser --> GradPadParser
+deactivate SearchCommandParser
+
+GradPadParser -> SearchCommandParser : parse("CS2103T")
+activate SearchCommandParser
+
+SearchCommandParser -> ParserUtil : parseModuleCode("CS2103T")
+activate ParserUtil
+
+ParserUtil --> SearchCommandParser : code
+deactivate ParserUtil
+
+create SearchCommand
+SearchCommandParser -> SearchCommand
+activate SearchCommand
+
+SearchCommand --> SearchCommandParser : s
+deactivate SearchCommand
+
+SearchCommandParser --> GradPadParser : s
+deactivate SearchCommandParser
+'Hidden arrow to position the destroy marker below the end of the activation bar.
+SearchCommandParser -[hidden]-> GradPadParser
+destroy SearchCommandParser
+
+GradPadParser --> LogicManager : s
+deactivate GradPadParser
+
+LogicManager -> SearchCommand : execute()
+activate SearchCommand
+
+create ModuleInfoSearcher
+SearchCommand -> ModuleInfoSearcher
+activate ModuleInfoSearcher
+
+ModuleInfoSearcher --> SearchCommand
+deactivate ModuleInfoSearcher
+
+SearchCommand -> ModuleInfoSearcher : searchModule()
+activate ModuleInfoSearcher
+
+ModuleInfoSearcher --> SearchCommand : r
```

Comment:
maybe we can change the "r" here to "module info" or something?

Reply:
Yep sounds good!